Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -0,0 +1,299 @@
+﻿/*
+Copyright 2015 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Google.Apis.Auth.OAuth2.Responses;
+using Google.Apis.Http;
+using System.IO;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Credential for authorizing calls using OAuth 2.0.
+    /// It is a convenience wrapper that allows handling of different types of 
+    /// credentials (like ServiceAccountCredential, ComputeCredential or UserCredential)
+    /// in a unified way.
+    /// <para>
+    /// See <see cref="GetApplicationDefaultCredential"/>.
+    /// </para>
+    /// </summary>
+    public abstract class GoogleCredential : IConfigurableHttpClientInitializer, ITokenAccess
+    {
+        private static DefaultCredentialProvider defaultCredentialProvider = new DefaultCredentialProvider();
```

Comment:
default provider  has the implementation  of GetApplicationDefaultAsync(). As we discussed before, it is useful to have default provider as a separate class because it makes testing easier.